pkg/btree/btree.go

```diff
@@ -236,6 +236,88 @@ func (s *children) truncate(index int) {
 	}
 }
 
+// size stores size info of a node.
+// size[i] is the rank of the max item of children[i] in the subtree, so we have following formulas:
+//
+//   size[0] = if i == 0 { children[0].length() }
+//             else { size[i-1] + 1 + children[i].length() }
+type size []int
+
+func (s *size) addAt(index int, delta int) {
+	for i := index; i < len(*s); i++ {
+		(*s)[i] += delta
```

Comment:
Can we abstract int to an interface like
```
type AttachInfo interface {
   Add(Item)
} 
``` 
so this can change to 
```
(*s)[i].Add(item)
```
What I mean a way to attach more statistics, like `write bytes`.

Reply:
I have thought about it, but didn't find a good way. Maybe we can do it in another PR?

Reply:
👍